tests/components/azure_data_explorer/conftest.py

```diff
@@ -0,0 +1,138 @@
+"""Test fixtures for ADX."""
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.azure_data_explorer.const import (
+    CONF_FILTER,
+    CONF_SEND_INTERVAL,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import STATE_ON
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import utcnow
+
+from .const import (
+    AZURE_DATA_EXPLORER_PATH,
+    BASE_CONFIG_FREE,
+    BASE_CONFIG_FULL,
+    BASIC_OPTIONS,
+)
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# fixtures for both init and config flow tests
+@dataclass
+class FilterTest:
+    """Class for capturing a filter test."""
+
+    entity_id: str
+    expected_count: int
+
+
+@pytest.fixture(name="filter_schema")
+def mock_filter_schema():
+    """Return an empty filter."""
+    return {}
+
+
+@pytest.fixture(name="entry_managed")
+async def mock_entry_fixture_managed(hass, filter_schema):
+    """Create the setup in HA."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=BASE_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    await _entry(hass, filter_schema, entry)
+    return entry
+
+
+@pytest.fixture(name="entry_queued")
+async def mock_entry_fixture_queued(hass, filter_schema):
+    """Create the setup in HA."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=BASE_CONFIG_FREE,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    await _entry(hass, filter_schema, entry)
+    return entry
+
+
+async def _entry(hass, filter_schema, entry):
+    entry.add_to_hass(hass)
+    assert await async_setup_component(
+        hass, DOMAIN, {DOMAIN: {CONF_FILTER: filter_schema}}
+    )
+    assert entry.state == ConfigEntryState.LOADED
+
+    # Clear the component_loaded event from the queue.
+    async_fire_time_changed(
+        hass,
+        utcnow() + timedelta(seconds=entry.options[CONF_SEND_INTERVAL]),
+    )
+    await hass.async_block_till_done()
+
+
+@pytest.fixture(name="entry_with_one_event")
+async def mock_entry_with_one_event(hass, entry_managed):
+    """Use the entry and add a single test event to the queue."""
+    assert entry_managed.state == ConfigEntryState.LOADED
+    hass.states.async_set("sensor.test", STATE_ON)
+    return entry_managed
+
+
+# Fixtures for config_flow tests
+@pytest.fixture(name="mock_setup_entry")
```

Comment:
```suggestion
@pytest.fixture
```

The name will be set automatically to the function name if omitted. So you don't need to set it, where it is the same as the function name. Please adopt the other fixtures too